servers/wildfly26.0/src/main/java/org/jboss/migration/wfly/task/security/MigrateLegacySecurityRealmsToElytron.java

```diff
@@ -78,172 +92,219 @@ protected MigrateToElytron(final LegacySecurityConfigurations legacySecurityConf
                 }
                 return ServerMigrationTaskResult.SUCCESS;
             };
             runBuilder(SubsystemResource.class, JBossSubsystemNames.ELYTRON, runnableBuilder);
         }
 
-        protected void migrateSecurityRealm(LegacySecurityRealm securityRealm, LegacySecurityConfiguration legacySecurityConfiguration, SubsystemResource subsystemResource, TaskContext taskContext) {
+        protected void migrateSecurityRealms(LegacySecurityConfiguration legacySecurityConfiguration, SubsystemResource subsystemResource, TaskContext taskContext) {
             final ManageableServerConfiguration configuration = subsystemResource.getServerConfiguration();
             final Operations.CompositeOperationBuilder compositeOperationBuilder = Operations.CompositeOperationBuilder.create();
-            addOperationSteps(securityRealm, legacySecurityConfiguration, subsystemResource, compositeOperationBuilder, taskContext);
+            addOperationSteps(legacySecurityConfiguration, subsystemResource, compositeOperationBuilder, taskContext);
             final ModelNode migrateOp = compositeOperationBuilder.build().getOperation();
-            taskContext.getLogger().debugf("Legacy security realm %s migration to Elytron's management op: %s", securityRealm.getName(), migrateOp);
+            taskContext.getLogger().debugf("Legacy security realms migration to Elytron's management op: %s", migrateOp);
             configuration.executeManagementOperation(migrateOp);
-            taskContext.getLogger().infof("Legacy security realm %s migrated to Elytron.", securityRealm.getName());
+            taskContext.getLogger().infof("Legacy security realms migrated to Elytron.");
         }
 
-        protected void addOperationSteps(LegacySecurityRealm securityRealm, LegacySecurityConfiguration legacySecurityConfiguration, SubsystemResource subsystemResource, Operations.CompositeOperationBuilder compositeOperationBuilder, TaskContext taskContext) {
-            if (securityRealm.getAuthentication() != null && securityRealm.getAuthorization() != null) {
-                addSecurityRealm(securityRealm, legacySecurityConfiguration, subsystemResource, compositeOperationBuilder, taskContext);
-                addSecurityDomain(securityRealm, legacySecurityConfiguration, subsystemResource, compositeOperationBuilder, taskContext);
-                addHttp(securityRealm, legacySecurityConfiguration, subsystemResource, compositeOperationBuilder, taskContext);
-                addSasl(securityRealm, legacySecurityConfiguration, subsystemResource, compositeOperationBuilder, taskContext);
-            }
-            addServerIdentities(securityRealm, legacySecurityConfiguration, subsystemResource, compositeOperationBuilder, taskContext);
+        protected void addOperationSteps(LegacySecurityConfiguration legacySecurityConfiguration, SubsystemResource subsystemResource, Operations.CompositeOperationBuilder compositeOperationBuilder, TaskContext taskContext) {
```

Comment:
The default resources are added even when not needed.

Reply:
This is intentional, I thought about it and it requires a lot of logic at XML level to figure out when it's needed or not (e.g. missing attributes which defaults to a legacy security realm or domain). Those resources need to be explicitly referenced to be used so I don't think there is any issue with always creating them. If you don't agree please open a new bug JIRA.